### PR TITLE
Add support for "belongsTo" relationships for create forms

### PR DESCRIPTION
**Pull request by lyonsbp**
*Issue #, if available:*

*Description of changes:*
This change supports "belongsTo" relationships for forms, putting an `<Autocomplete>` field into the form allowing one related model to be selected.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-golden-files/lib/react/forms/datastore-create-form-with-belongs-to/MemberCreateForm.jsx [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+          }, []),
+        );
+        if (validationResponses.some((r) => r.hasError)) {
+          return;
+        }
+        if (onSubmit) {
+          modelFields = onSubmit(modelFields);
+        }
+        try {
+          await DataStore.save(new Member(modelFields));
```
Comment (zchenwei) [inline packages/codegen-ui-golden-files/lib/react/forms/datastore-create-form-with-belongs-to/MemberCreateForm.jsx:239]: I could be wrong but would this be necessary create a bi-directional relationship without doing anything to the team? Like,
```
await DataStore.save(Team.copyOf(team, team => team.Members.push[newMember]));
```

File: packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react-forms.test.ts [hunk 2]
```diff
@@ -58,6 +58,21 @@ describe('amplify form renderer tests', () => {
       expect(declaration).toMatchSnapshot();
     });
 
+    it('should generate a create form with belongsTo relationship', () => {
+      const { componentText, declaration } = generateWithAmplifyFormRenderer(
+        'forms/member-datastore-create',
+        'datastore/project-team-model',
+      );
+      // check nested model is imported
```
Comment (cwoolum) [inline packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react-forms.test.ts:66]: Can we remove this commented out code?

File: packages/codegen-ui/lib/generate-form-definition/helpers/model-fields-configs.ts [hunk 3]
```diff
@@ -69,15 +69,11 @@ function getValueMappings({
 
   // if relationship
   if (field.relationship) {
-    // if model & HAS_ONE
-    if (field.relationship.type === 'HAS_ONE') {
-      const modelName = field.relationship.relatedModelName;
-      return {
-        // TODO: map field dynamically as part of cpk task
```
Comment (zchenwei) [inline packages/codegen-ui/lib/generate-form-definition/helpers/model-fields-configs.ts:76]: We probably want to keep this TODO

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/all-props.ts [hunk 4]
```diff
@@ -72,9 +72,9 @@ export const addFormAttributes = (component: StudioComponent | StudioComponentCh
     }
 
     // TODO: Allow for other relationship types once valueMappings available
```
Comment (zchenwei) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper/all-props.ts:74]: We may want to remove this TODO?

File: packages/codegen-ui-react/lib/forms/form-renderer-helper/event-handler-props.ts [hunk 5]
```diff
@@ -282,12 +282,12 @@ export const buildOnChangeStatement = (
 
 /**
   example:
-  onSuggestionSelect={({ id, label }) => {
+  onSelect={({ id, label }) => {
```
Comment (zchenwei) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper/event-handler-props.ts:285]: You probably want to update the above comment as well

File: packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react-forms.test.ts (hunk 2 above)
Comment (zchenwei) [reply packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react-forms.test.ts:66]: +1

**Review by cwoolum — DISMISSED** [review]

**Review by cwoolum — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

**Pull request merged**